test/tools/fuzzTesting/createRandomTest.cpp

```diff
@@ -185,84 +146,92 @@ void fillRandomTest(std::function<void(json_spirit::mValue&, bool)> _doTests, st
-			else
-			if (types.at(i) == "[0xHASH32]")
-				_test.replace(pos, 10, "0x" + dev::test::RandomCode::rndByteSequence(32));
-			else
-			if (types.at(i) == "[HASH32]")
-				_test.replace(pos, 8, dev::test::RandomCode::rndByteSequence(32));
-			else
-			if (types.at(i) == "[V]")
+			else if (type == "[CODE]")
+			{
+				int random = (int)dev::test::RandomCode::randomUniInt() % 100;
+				if (random < 90)
+					_test.replace(pos, type.length(), "0x"+dev::test::RandomCode::generate(5, options));
+				else
+					_test.replace(pos, type.length(), "");
+			}
+			else if (type == "[HEX]")
+				_test.replace(pos, type.length(), dev::test::RandomCode::randomUniIntHex());
+			else if (type == "[HEX32]")
+			{
+				int random = (int)dev::test::RandomCode::randomUniInt() % 100;
+				if (random < 90)
+					_test.replace(pos, type.length(), dev::test::RandomCode::randomUniIntHex(std::numeric_limits<uint32_t>::max()));
+				else
+					_test.replace(pos, type.length(), "0x00");
+			}
+			else if (type == "[GASLIMIT]")
+				_test.replace(pos, type.length(), dev::test::RandomCode::randomUniIntHex(dev::u256("36028797018963967"), dev::u256("18014398509481983")));
+			else if (type == "[GASPRICE]")
+				_test.replace(pos, type.length(), dev::test::RandomCode::randomUniIntHex(dev::u256("5")));
+			else if (type == "[HASH20]")
+				_test.replace(pos, type.length(), dev::test::RandomCode::rndByteSequence(20));
+			else if (type == "[ADDRESS]")
+				_test.replace(pos, type.length(), toString(options.getRandomAddress()));
+			else if (type == "[0xADDRESS]")
+				_test.replace(pos, type.length(), "0x" + toString(options.getRandomAddress()));
+			else if (type == "[0xDESTADDRESS]")
+			{
+				int random = (int)dev::test::RandomCode::randomUniInt() % 100;
+				if (random < 50)
+					_test.replace(pos, type.length(), "0x" + toString(options.getRandomAddress()));
+				else
+					_test.replace(pos, type.length(), "");
+			}
+			else if (type == "[0xHASH32]")
+				_test.replace(pos, type.length(), "0x" + dev::test::RandomCode::rndByteSequence(32));
+			else if (type == "[HASH32]")
+				_test.replace(pos, type.length(), dev::test::RandomCode::rndByteSequence(32));
+			else if (type == "[V]")
 			{
 				int random = (int)dev::test::RandomCode::randomUniInt() % 100;
 				if (random < 30)
-					_test.replace(pos, 3, "0x1c");
+					_test.replace(pos, type.length(), "0x1c");
 				else
 				if (random < 60)
-					_test.replace(pos, 3, "0x1d");
+					_test.replace(pos, type.length(), "0x1d");
 				else
-					_test.replace(pos, 3, "0x" + dev::test::RandomCode::rndByteSequence(1));
+					_test.replace(pos, type.length(), "0x" + dev::test::RandomCode::rndByteSequence(1));
 			}
 			else
-				_test.replace(pos, types.at(i).size(), _varMap.at(types.at(i)));
+			{
+				//Replace type from varMap if varMap is set
+				if (_varMap.count(type))
+					_test.replace(pos, type.length(), _varMap.at(type));
+				BOOST_ERROR("Skipping undeclared type: " + type);
+			}
 
```

Comment:
You can siplify the code by defining `string replacement`.  In each case, you just do `replacement = something;`

After all these cases, a single
```
_test.replace(pos, type.length(), replacement);
```
will do the job.  This way, the many repetitions of `_test.replace(pos, type.length(), ....);` can be removed.